pallets/task-scheduler/src/lib.rs

```diff
@@ -99,6 +106,76 @@ pub mod pallet {
 		/// callback transaction. This is likely an internal error.
 		NoLocalAcctForSignedTx,
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(block_number: T::BlockNumber) -> Weight {
+			use sp_io::KillStorageResult;
+
+			log::debug!("Cleaning up expired entries");
+
+			let mut unverified_task_count = 0u32;
+			loop {
+				match PendingTasks::<T>::remove_prefix(block_number, None) {
+					KillStorageResult::SomeRemaining(u) => {
+						unverified_task_count.saturating_accrue(u)
+					},
+					KillStorageResult::AllRemoved(u) => {
+						unverified_task_count.saturating_accrue(u);
+						break;
+					},
+				};
+			}
+
+			<T as Config>::WeightInfo::on_initialize(unverified_task_count)
+		}
+
+		fn offchain_worker(block_number: T::BlockNumber) {
+			let auth_id = match Self::authority_id() {
```

Comment:
Hmm so I'm kind of split on whether the concept of authorities belong in this pallet.

If we want to keep the scope of the pallet to just task scheduling, I think authorities shouldn't be included and they should stay in the creditcoin pallet for now. Authorities are more of an implementation detail in our off-chain verification scheme, and they don't really pertain to task scheduling. I could imagine having a chain where we don't use the authorities scheme, and we would have to modify this pallet (taking authorities out) to use it on that new chain, which I think isn't ideal. Ideally, we should only have to touch this pallet if we're changing the task scheduling logic (like how we handle the task guard, the nonce management, or adding a fancier scheduling algorithm). And if we wanted to change how our authority model works, it would be weird to have to change the "task scheduling" pallet to do that.

On the other hand, I get your point that the more we remove from this pallet, the less useful it seems. I think then if we want to expand the scope of this pallet (to include, e.g., authorities), we should rename/reframe the purpose of the pallet - changing it from just task scheduling to a more general ocw pallet.

What are your thoughts?

Reply:
I agree, I avoided overthinking what to do with the 'authority' for now. I can't think of another use case for the pallet so far other than what the currently have; coming with an interface for the signer is not trivial atm. I was thinking about providing an interface for the 'signer' to decouple signing from the task-scheduling; the main advantage being 'security' (maybe?, not exposing keyrings to the pallet) and also decoupling 'authorities' logic from the signing logic.

Reply:
Hmm yeah I think it's okay for now. We actually might want to break out an authorities pallet at some point, once the logic gets more complicated (staking and slashing and all that)